File: JANUARY/26-01-2026/1200.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        sort(arr.begin(), arr.end());
        int mn = INT_MAX;
        for (int i = 1; i < arr.size(); i++)
            mn = min(mn, arr[i] - arr[i - 1]);
        
        vector<vector<int>> res;
        for (int i = 1; i < arr.size(); i++)
            if (arr[i] - arr[i - 1] == mn)
                res.push_back({arr[i - 1], arr[i]});
        
        return res;
    }
};

File: JANUARY/26-01-2026/1200.rs
impl Solution {
    pub fn minimum_abs_difference(mut arr: Vec<i32>) -> Vec<Vec<i32>> {
        arr.sort();
        let mut mn = i32::MAX;
        for i in 1..arr.len() {
            mn = mn.min(arr[i] - arr[i - 1]);
        }
        let mut res = Vec::new();
        for i in 1..arr.len() {
            if arr[i] - arr[i - 1] == mn {
                res.push(vec![arr[i - 1], arr[i]]);
            }
        }
        res
    }
}